docs/taskcluster-yml.md

```diff
@@ -0,0 +1,91 @@
+---
+title: .taskcluster.yml
+---
+
+When a push occurs on a monitored repository, the service looks for a
+`.taskcluster.yml` in the root of the repository at the pushed commit. That
+file is read as a YAML document, rendered with
+[JSON-e](https://github.com/taskcluster/json-e), and then used to create the
+resulting tasks.
+
+## File Versions
+
+Every template must contain a top-level property `version: 1`. This is
+consulted before the JSON-e rendering takes place.
+
+There was a version 0 template format, but it was undocumented and was not
+valid YAML.  We shall speak of it no more.
+
+## JSON-e Rendering
+
+The entire YAML file is rendered using
+[JSON-e](https://github.com/taskcluster/json-e). The following context
+variables are provided:
+
+```js
+{
+  // tasks_for (string) - always 'hg-push' for mozilla-taskcluster (this is to
+  // distinguish from other tools that might read the same template, such as
+  // taskcluster-github)
+  tasks_for: 'hg-push',
+
+  push: {
+    // push.owner (string) - the hg user that made the push (not always an email)
+    owner: '..',
```

Comment:
I think it would be best to do the transformation to email on the mozilla-taskcluster side?

But I could be wrong, given that the goal here is reproducibility in CoT verification...